model.py

```diff
@@ -0,0 +1,483 @@
+"""Model for the NFA ShotGrid Project Creator, written by Mervin van Brakel (2024)"""
+
+from __future__ import annotations
+
+import datetime
+import os
+import re
+from dataclasses import dataclass
+
+from shotgun_api3 import shotgun
+
+
+@dataclass
+class ProjectInformation:
+    """Dataclass used for storing all the information we need to create the project."""
+
+    username: str
+    project_name: str
+    has_production_code: bool
+    project_code: str
+    has_other_supervisors: bool
+    supervisor_list: list[dict]
+    render_engine: str
+    project_type: str
+    project_fps: int
+
+
+@dataclass
+class UserInformation:
+    """Dataclass used for storing information about the user who creates the project."""
+
+    sg_username: str
+    sg_user_id: str
+    student_year: int
+    student_graduation_year: int
+
+
+class ProjectCreatorModel:
```

Comment:
Lets split this entire class into functions.

Then I can imagine "but how about storing the sg connection? well, we can add a lru_cache, this will store the connection on a function. Lets have a call soon, then we can check it out :)